src/overlays/actors/ovl_Item_Shield/z_item_shield.c

```diff
@@ -138,38 +147,51 @@ void func_80B86BC8(ItemShield* this, PlayState* play) {
     }
 }
 
-void func_80B86CA8(ItemShield* this, PlayState* play) {
-    static Vec3f D_80B871F4 = { 0.0f, 0.0f, 0.0f };
-    static f32 D_80B87200[] = { 0.3f, 0.6f,  0.9f, 1.0f,  1.0f, 1.0f,  1.0f, 1.0f,
-                                1.0f, 0.85f, 0.7f, 0.55f, 0.4f, 0.25f, 0.1f, 0.0f };
-    static f32 D_80B87240[] = { 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.8f,
-                                0.6f, 0.4f, 0.2f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+#define ITEMSHIELD_FLAME_TYPE_COUNT 15
+
+void ItemShield_Burning_Combust(ItemShield* this, PlayState* play) {
+    static Vec3f sFlamePos = { 0.0f, 0.0f, 0.0f };
+    static f32 sFlameScales[] = {
+        0.3f, 0.6f, 0.9f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.85f, 0.7f, 0.55f, 0.4f, 0.25f, 0.1f, 0.0f,
+    };
+    static f32 sFlameIntensities[] = {
+        1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 1.0f, 0.8f, 0.6f, 0.4f, 0.2f, 0.0f, 0.0f, 0.0f, 0.0f, 0.0f,
+    };
     s32 i;
-    s32 temp;
+    s32 type;
 
     Actor_MoveXZGravity(&this->actor);
     Actor_UpdateBgCheckInfo(play, &this->actor, 10.0f, 10.0f, 0.0f, UPDBGCHECKINFO_FLAG_0 | UPDBGCHECKINFO_FLAG_2);
     this->actor.shape.yOffset = ABS(Math_SinS(this->actor.shape.rot.x)) * 1500.0f;
 
-    for (i = 0; i < 8; i++) {
-        temp = 15 - this->unk_19E[i];
-        D_80B871F4.x = this->unk_1A8[i].x;
-        D_80B871F4.y = this->unk_1A8[i].y + (this->actor.shape.yOffset * 0.01f) + (D_80B87200[temp] * -10.0f * 0.2f);
-        D_80B871F4.z = this->unk_1A8[i].z;
-        EffectSsFireTail_SpawnFlame(play, &this->actor, &D_80B871F4, D_80B87200[temp] * 0.2f, -1, D_80B87240[temp]);
-        if (this->unk_19E[i] != 0) {
-            this->unk_19E[i]--;
+    // Spawn flames at random locations until despawn
+    for (i = 0; i < ITEMSHIELD_FLAME_COUNT; i++) {
+        type = ITEMSHIELD_FLAME_TYPE_COUNT - this->flameTypes[i];
+
+        sFlamePos.x = this->flameBasePoss[i].x;
+        sFlamePos.y =
+            this->flameBasePoss[i].y + (this->actor.shape.yOffset * 0.01f) + (sFlameScales[type] * -10.0f * 0.2f);
+        sFlamePos.z = this->flameBasePoss[i].z;
+
+        EffectSsFireTail_SpawnFlame(play, &this->actor, &sFlamePos, sFlameScales[type] * 0.2f, -1,
+                                    sFlameIntensities[type]);
+
+        // cycle through flame types and pick new random positions until the shield is about to disappear
+        if (this->flameTypes[i] != 0) {
+            this->flameTypes[i]--;
         } else if (this->timer > 16) {
-            this->unk_19E[i] = 15;
-            this->unk_1A8[i].x = Rand_CenteredFloat(15.0f);
-            this->unk_1A8[i].y = Rand_CenteredFloat(10.0f);
-            this->unk_1A8[i].z = Rand_CenteredFloat(15.0f);
+            this->flameTypes[i] = ITEMSHIELD_FLAME_TYPE_COUNT;
+            this->flameBasePoss[i].x = Rand_CenteredFloat(15.0f);
+            this->flameBasePoss[i].y = Rand_CenteredFloat(10.0f);
+            this->flameBasePoss[i].z = Rand_CenteredFloat(15.0f);
```

Comment:
the plural s on `Pos` looks funny. `this->flameBasePos` is probably fine, but could also do `this->flameBasePosList` or `this->flameBasePosArr`?

Reply:
went with something a bit different, see how that looks

Reply:
actually, its more of an offset than anything
the effect handles the "attaching to body parts" part (in a draw function no less)

Reply:
this kinda rabbit holes into a question of "should the effect refer to this as an offset everywhere, and break the normal pattern of pos/vel/accel"

maybe just leave it as pos for now and let an effects focused pass answer this question

Reply:
Yeah, that's fine with me